File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/Thrift/src/main/java/root/proiect_mpp/service/logIn/LogInService.java
package root.proiect_mpp.service.logIn;

import root.proiect_mpp.domain.people.Employee;
import root.proiect_mpp.repositories.people.employees.EmployeeRepository;

public class LogInService implements ILogInService {
    private final EmployeeRepository employeeRepository;

    public LogInService(EmployeeRepository employeeRepository) {
        this.employeeRepository = employeeRepository;
    }

    public Employee findByEmail(String email) {
        // Validate email
        return employeeRepository.findByEmail(email);
    }
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/Proiect_MPP/service/logIn/LogInService.cs
using Proiect_MPP.domain.people;
using Proiect_MPP.repository.people.employees;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Proiect_MPP.service.logIn
{
    public class LogInService : ILogInService
    {
        private readonly IEmployeeRepository _employeeRepository;
        public LogInService(IEmployeeRepository employeeRepository)
        {
            _employeeRepository = employeeRepository;
        }
        public Employee findByEmail(string email)
        {
            return _employeeRepository.findByEmail(email);
        }
    }
}
